src/reverb.js

```diff
@@ -268,37 +268,280 @@ p5.Reverb.prototype.dispose = function () {
  *  }
  *  </code></div>
  */
-p5.Convolver = function (path, callback, errorCallback) {
-  p5.Reverb.call(this);
+class Convolver extends Reverb {
+  constructor(path, callback, errorCallback) {
+    super();
+    /**
+     *  Internally, the p5.Convolver uses the a
+     *  <a href="http://www.w3.org/TR/webaudio/#ConvolverNode">
+     *  Web Audio Convolver Node</a>.
+     *
+     *  @property {ConvolverNode} convolverNode
+     */
+    this._initConvolverNode();
+
+    // otherwise, Safari distorts
+    this.input.gain.value = 0.5;
+
+    if (path) {
+      this.impulses = [];
+      this._loadBuffer(path, callback, errorCallback);
+    } else {
+      // parameters
+      this._seconds = 3;
+      this._decay = 2;
+      this._reverse = false;
+
+      this._buildImpulse();
+    }
+
+    /**
+     *  If you load multiple impulse files using the .addImpulse method,
+     *  they will be stored as Objects in this Array. Toggle between them
+     *  with the <code>toggleImpulse(id)</code> method.
+     *
+     *  @property {Array} impulses
+     *  @for p5.Convolver
+     */
+    this.impulses = [];
+    this.set = null;
+  }
 
   /**
-   *  Internally, the p5.Convolver uses the a
-   *  <a href="http://www.w3.org/TR/webaudio/#ConvolverNode">
-   *  Web Audio Convolver Node</a>.
+   *  Private method to load a buffer as an Impulse Response,
+   *  assign it to the convolverNode, and add to the Array of .impulses.
    *
-   *  @property {ConvolverNode} convolverNode
+   *  @param   {String}   path
+   *  @param   {Function} callback
+   *  @param   {Function} errorCallback
+   *  @private
    */
-  this._initConvolverNode();
+  _loadBuffer(_path, callback, errorCallback) {
+    var path = p5.prototype._checkFileFormats(_path);
```

Comment:
same comment as here https://github.com/processing/p5.js-sound/pull/531#discussion_r475322629 (and made a card for this one in our backlog https://trello.com/c/2fweJih8/51-remove-checkfileformats-from-p5prototype-and-import-it-instead)